app/full_text_search.py

```diff
@@ -106,16 +109,78 @@ def search(repo, query_txt, max_results):
             results.append(result)
     return results
 
-def create_document(**kwargs):
+
+def create_jp_name_document(**kwargs):
```

Comment:
Can you move this function into jautil? Ideally, full_text_search.py shouldn't include much of language-depedent logic.